bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/InsteonBinding.java

```diff
@@ -125,6 +125,12 @@ public InsteonBinding(InsteonNetworkHandler handler, @Nullable InsteonNetworkCon
             @Nullable SerialPortManager serialPortManager) {
         this.handler = handler;
 
+        String port = config.getPort();
+        logger.debug("port = '{}'", Utils.redactPassword(port));
+
+        driver = new Driver(port, portListener, serialPortManager);
+        driver.addMsgListener(portListener);
```

Comment:
Any reason this is no longer evaluated at the end of the constructor? You should make sure not the leak any references of `this` until after you have finished essential configuration.

Reply:
Driver previously didn't have a constructor and was created when driver was declared above.  This change allowed the removal of `@SuppressWarnings("null")` from Driver by passing them in as parameters to the constructor instead of using setters.

Reply:
What I meant is that this new code was put at the beginning of the InsteonBinding constructor whereas the old code was at the end of the InsteonBinding constructor.

Reply:
Yes that's because the old code had this above:

`private Driver driver = new Driver();`

Why does it matter? Where is this leaked?

Reply:
`this` is leaked through the PortListener.

Reply:
I still don't understand the concern, and why it matters. If it's a big deal, maybe I'll just delete this PR and stop contributing.

Reply:
The general concern over a leak of `this` is that other might start calling methods on your InsteonBinding class when it hasn't even finished constructing. What would happen if PortListener starts getting call on its methods before you have finished loading devices? The instant you leak `this` in a constructor you put yourself at risk of bugs NPE bugs that would normally not be possible if an object was constructed completely before exposing it to other objects.

In this case it might not cause a problem since things shouldn't start calling PortListener until you call `driver.start()` but leaking the `this` reference is a general problem to be aware of.

That said leaking the `this` of often the most convenient way to code something so at the very least try to make sure you leak `this` at the very end of your constructor when you know all the other class fields have been initialized, thus `this` would be "safe" to call from outside classes.

http://www.javapractices.com/topic/TopicAction.do?Id=252

Reply:
You might notice from the link I posted that they suggest not leaking `this` at ALL. While I agree with it in principle, I also think that trying to enforce that would cause a bit a code bloat so that is why I recommend just leaking `this` at the end of the constructor.

Reply:
@kaikreuzer, this is ridiculous, lot of assumptions here. I thought I was doing a good thing by removing `@SuppressWarnings("null")` from Driver and rearranging the code, but I guess not according to @cpmeister.
This sort of nitpicking is what causes people to stop contributing, and I'm about to that point.  I have some other features that I was thinking about implementing, but probably won't if this continues.

Reply:
@robnielsen I am very sure that @cpmeister is not meaning to annoy you in any way. Like you and others, he wants to help and ensure that openHAB has a proper code base.
Nobody should be felt pissed off by anyone else, when everybody is doing their best - so please consider this and try to solve any discussions in a collaborative manner. You should be aware that many developers LOVE such discussions, because they learn from each other and become better developers in the long run - it is part of the fun. If it is no fun for you, just express it or simply give in to the arguments.